src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -1034,3 +1016,17 @@ func isRedpandaClusterVersionManaged(
 	}
 	return true
 }
+
+func (r *ClusterReconciler) adminAPIClient(
+	ctx context.Context,
+	cluster *redpandav1alpha1.Cluster,
+	fqdn string,
+	adminTLSProvider resourcetypes.AdminTLSConfigProvider,
+	ordinals ...int32,
+) (adminutils.AdminAPIClient, error) {
+	aa, err := r.AdminAPIClientFactory(ctx, r.Client, cluster, fqdn, adminTLSProvider, ordinals...)
+	if err != nil {
+		return nil, err
+	}
+	return aa, nil
+}
```

Comment:
NIT: I don't understand why this function is created. What this function gives?

Reply:
Updated